plugins/orchestrator-backend/src/service/SonataFlowService.ts

```diff
@@ -53,48 +52,54 @@ interface JiraConfig {
 
 export class SonataFlowService {
   private readonly connection: SonataFlowConnectionConfig;
+  private dataIndex: DataIndexService;
 
   constructor(
     config: Config,
+    dataIndexService: DataIndexService,
     private readonly logger: Logger,
   ) {
     this.connection = this.extractConnectionConfig(config);
+    this.dataIndex = dataIndexService;
   }
 
   public get autoStart(): boolean {
     return this.connection.autoStart;
   }
 
-  public get url(): string {
-    if (!this.connection.port) {
-      return this.connection.host;
-    }
-    return `${this.connection.host}:${this.connection.port}`;
-  }
-
   public get resourcesPath(): string {
     return this.connection.resourcesPath;
   }
 
   public async connect(): Promise<boolean> {
-    const isAlreadyUp = await this.isSonataFlowUp(false);
-    if (isAlreadyUp) {
+    if (!!this.connection.autoStart) {
```

Comment:
No need for `!!` as `autoStart` is already a boolean and can't be `undefined`.
Here you are basically saying `if (this.connection.autoStart)`
And, btw, if `autoStart` is true, then this code is not reaching `this.launchSonataFlow()` anymore.